### PR TITLE
Feature/ us 0007

### LINKED ISSUE #13: US - 0007
Criação da rota de unfollow para um vendedor e implementação do serviço.

### PR DESCRIPTION
**Pull request by lmarizmeli**
- [x] Created route to unfollow
- [x] Created method Customer.removeFollow

### CODE REVIEW

**Review by Victor-Planas — APPROVED** [review]
Gorgeous!! :trollface:

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java [hunk 1]
```diff
@@ -61,4 +62,13 @@ public GenericMessageDTO followSeller(@PathVariable @Min(0) Integer userId, @Pat
         usersService.follow(userId,userIdToFollow);
         return new GenericMessageDTO("Seller followed succesfully!");
     }
+
+    @PostMapping("/{userId}/unfollow/{userIdToUnfollow}")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "unfollows customer to seller, userId refers to customer and userIdToUnfollow refers to seller\n")
```
Comment (CarlosFeldmann) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java]: Starts with upper case. And i didn't get the new line at the end

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/entities/Customer.java [hunk 2]
```diff
@@ -29,6 +30,18 @@ public void addFollow(Seller seller){
...
 
+    /**
+     *
+     * @param seller - seller who will be unfollowed
+     */
+    public void removeFollow(Seller seller){
+        FollowDate followMatch = followed.stream()
+                .filter(a -> a.getSeller().equals(seller))
+                .findFirst()
+                .orElseThrow(() -> new ResourceNotFoundException("Seller", seller.getUserId()));
```
Comment (CarlosFeldmann) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/entities/Customer.java]: The resource in ResourceNotFoundException here is the FollowDate and not Seller

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/services/UsersService.java [hunk 3]
```diff
@@ -85,5 +85,24 @@ public void follow(Integer customerId, Integer sellerId) {
...
+     * @param sellerId - seller Id
+     */
+    public void unfollow(Integer customerId, Integer sellerId) {
+        Customer customer = customerRepository.findById(customerId)
+                .orElseThrow(() -> new ResourceNotFoundException("Customer", customerId));
+        Seller seller = sellerRepository.findById(sellerId)
+                .orElseThrow(() -> new ResourceNotFoundException("Seller", sellerId));
+
+        if (!customer.isFollowing(seller))
+            throw new ApiException(HttpStatus.BAD_REQUEST, "already_unfollows_error", "User already unfollows seller");
```
Comment (CarlosFeldmann) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/services/UsersService.java]: Already unfollows is strange, i think customer_isnt_following_error, something like that

**Review by CarlosFeldmann — CHANGES_REQUESTED** [review]

**Comment by CarlosFeldmann** [comment]
🦧

**Review by Giovannapls — APPROVED** [review]
🤝

**Review by CarlosFeldmann — APPROVED** [review]
Nice 🦧🦧🦧🦧🦧🦧🦧

**Review by CarlosFeldmann — APPROVED** [review]

**Comment by CarlosFeldmann** [comment]
🦧🦧🦧🦧

**Pull request merged**